hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/HandleContext.java

```diff
@@ -114,21 +132,32 @@ enum TransactionCategory {
      * there will be no corresponding {@link SignatureVerification}. If the key was provided during pre-handle, then the
      * corresponding {@link SignatureVerification} will be returned with the result of that verification operation.
      *
+     * <p>The signatures of required keys are guaranteed to be verified. Optional signatures may still be in the
+     * process of being verified (and therefore may time out). The timeout can be configured via the configuration
+     * {@code hedera.workflow.verificationTimeoutMS}
+     *
      * @param key the key to get the verification for
-     * @return the verification for the given key, or {@code null} if no such key was provided during pre-handle
+     * @return the verification for the given key
      * @throws NullPointerException if {@code key} is {@code null}
+     * @throws TimeoutException if the verification for the given key timed out
      */
-    @Nullable
-    SignatureVerification verificationFor(@NonNull Key key);
+    @NonNull
+    SignatureVerification verificationFor(@NonNull Key key) throws TimeoutException;
```

Comment:
The timeout value is configured centrally.

What should we do instead of throwing an exception? If we run into a timeout, the node is most likely going to ISS anyway. Maybe the right thing is to throw an Error and kill the node.

Reply:
As discussed, we log a warning and return a failing `SignatureVerification`. If all nodes run into a timeout, we can continue. If only one node times out, it will most likely ISS soon.